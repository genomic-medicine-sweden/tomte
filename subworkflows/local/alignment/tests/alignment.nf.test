nextflow_workflow {

    name "Test Workflow ALIGNMENT"
    script "../main.nf"
    workflow "ALIGNMENT"
    config "./nextflow.config"
    tag "subworkflow"
    tag "alignment"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
            script "../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'fasta' ],
                    file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("GUNZIP", alias: "GUNZIP_GTF") {
            script "../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'gtf' ],
                    file("${projectDir}/test_data/grch37_chr21.gtf.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("GUNZIP", alias: "GUNZIP_TRANSCRIPT") {
            script "../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'transcripts' ],
                    file("${projectDir}/test_data/grch37_chr21_transcripts.fa.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("STAR_GENOMEGENERATE") {
            script "../../../../modules/nf-core/star/genomegenerate/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = GUNZIP_GTF.out.gunzip
                """
            }
        }

        run("SALMON_INDEX") {
            script "../../../../modules/nf-core/salmon/index/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                    .map{ meta, fasta -> [ fasta ] }
                input[1] = GUNZIP_TRANSCRIPT.out.gunzip
                    .map{ meta, transcript -> [ transcript ] }
                """
            }
        }
    }

    test("fastq_alignment") {
        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ case:'finequagga', id:'ACC5963A2_id1', sample:'ACC5963A2', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:true ],
                    [ file("${projectDir}/test_data/ACC5963A1_sub_1.fastq.gz", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1_sub_2.fastq.gz", checkIfExists: true)]
                ])
                input[1]  = Channel.empty()
                input[2]  = STAR_GENOMEGENERATE.out.index
                input[3]  = GUNZIP_GTF.out.gunzip
                input[4]  = Channel.of(['illumina'])
                input[5]  = Channel.of([file("${projectDir}/test_data/subsample.bed", checkIfExists: true)])
                input[6]  = 0.001
                input[7]  = 120000000
                input[8]  = true
                input[9]  = true
                input[10] = SALMON_INDEX.out.index
                input[11] = GUNZIP_FASTA.out.gunzip
                input[12]  = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.versions,
                    workflow.out.bam_bai,
                    workflow.out.bam_ds_bai
                ).match() }
            )
        }

    }

    test("bam_alignment") {
        when {
            workflow {
                """
                input[0]  = Channel.empty()
                input[1]  = Channel.of([
                    [ case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:false ],
                    [ file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)]
                ])
                input[2]  = STAR_GENOMEGENERATE.out.index
                input[3]  = GUNZIP_GTF.out.gunzip
                input[4]  = Channel.of(['illumina'])
                input[5]  = Channel.of([file("${projectDir}/test_data/subsample.bed", checkIfExists: true)])
                input[6]  = 0.001
                input[7]  = 120000000
                input[8]  = true
                input[9]  = true
                input[10] = SALMON_INDEX.out.index
                input[11] = GUNZIP_FASTA.out.gunzip
                input[12]  = false
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.versions,
                    workflow.out.bam_bai,
                    workflow.out.bam_ds_bai
                ).match() }
            )
        }

    }

}
