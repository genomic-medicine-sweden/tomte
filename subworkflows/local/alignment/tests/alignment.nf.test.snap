{
    "bam_alignment": {
        "content": [
            [
                
            ],
            [
                [
                    {
                        "case": "finequagga",
                        "id": "ACC5963A1",
                        "sample": "ACC5963A1",
                        "strandedness": "reverse",
                        "sex": "NA",
                        "single_end": false,
                        "fq_pairs": 1,
                        "is_fastq": false
                    },
                    "ACC5963A1.bam:md5,0f3e153a88d96c4ad6a59898d2bea037",
                    "ACC5963A1.bam.bai:md5,4d81fe5964abc373dab11ca8eb098faa"
                ]
            ],
            [
                [
                    {
                        "case": "finequagga",
                        "id": "ACC5963A1",
                        "sample": "ACC5963A1",
                        "strandedness": "reverse",
                        "sex": "NA",
                        "single_end": false,
                        "fq_pairs": 1,
                        "is_fastq": false
                    },
                    "ACC5963A1.bam:md5,0f3e153a88d96c4ad6a59898d2bea037",
                    "ACC5963A1.bam.bai:md5,4d81fe5964abc373dab11ca8eb098faa"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T07:41:43.754107"
    },
    "fastq_alignment": {
        "content": [
            [
                [
                    {
                        "groupSize": 1,
                        "groupTarget": {
                            "case": "finequagga",
                            "id": "ACC5963A2",
                            "sample": "ACC5963A2",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        }
                    },
                    "ACC5963A2.Aligned.sortedByCoord.out.bam:md5,4d54d92054e87da392b4418daed05fe4",
                    "ACC5963A2.Aligned.sortedByCoord.out.bam.bai:md5,a5d4a2a3b7226aa89d8790c15be9c7e2"
                ]
            ],
            [
                [
                    {
                        "groupSize": 1,
                        "groupTarget": {
                            "case": "finequagga",
                            "id": "ACC5963A2",
                            "sample": "ACC5963A2",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        }
                    },
                    "ACC5963A2.Aligned.sortedByCoord.out.bam:md5,4d54d92054e87da392b4418daed05fe4",
                    "ACC5963A2.Aligned.sortedByCoord.out.bam.bai:md5,a5d4a2a3b7226aa89d8790c15be9c7e2"
                ]
            ],
            "ACC5963A2.cram.crai",
            "ACC5963A2.cram",
            {
                "maxReadLength": 151,
                "minReadLength": 13,
                "meanReadLength": 138,
                "maxQuality": 255,
                "minQuality": 0,
                "meanQuality": 253,
                "readCount": 209253
            },
            [
                "versions.yml:md5,26e766c12c867bec19ce7936855c9506",
                "versions.yml:md5,7c11e9e9e5618299c34aca60c92450a6",
                "versions.yml:md5,bd1f6a0d8ee30d9ad14843e2abfcf9e3",
                "versions.yml:md5,c6359df28f45236797d408fb562cef4d",
                "versions.yml:md5,fc3cdfeb9650e6068242aa84be1e3db1"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-07-29T15:01:03.375944904"
    }
}